### ISSUE TITLE
Round trip a Vega-Lite spec?

### ISSUE THREAD

**Issue by Symbolics** [yitzchak/shasht#2]
Should you be able to round-trip a vega-lite JSON spec? The use case is to find a simple DSL on the lisp side to encode a Vega-Lite spec with, and I thought I'd start by seeing how they decode and use them as examples to start with. However no joy. For example, using the [simple bar chart specification](https://vega.github.io/vega-lite/examples/bar.html) example on the Vega-Lite website:
```lisp
(defparameter *sbc*
"{
  \"$schema\": \"https://vega.github.io/schema/vega-lite/v5.json\",
  \"description\": \"A simple bar chart with embedded data.\",
  \"data\": {
    \"values\": [
      {\"a\": \"A\", \"b\": 28}, {\"a\": \"B\", \"b\": 55}, {\"a\": \"C\", \"b\": 43},
      {\"a\": \"D\", \"b\": 91}, {\"a\": \"E\", \"b\": 81}, {\"a\": \"F\", \"b\": 53},
      {\"a\": \"G\", \"b\": 19}, {\"a\": \"H\", \"b\": 87}, {\"a\": \"I\", \"b\": 52}
    ]
  },
  \"mark\": \"bar\",
  \"encoding\": {
    \"x\": {\"field\": \"a\", \"type\": \"nominal\", \"axis\": {\"labelAngle\": 0}},
    \"y\": {\"field\": \"b\", \"type\": \"quantitative\"}
  }
}")
```
Decode it as an alist:
```
(let ((shasht:*read-default-object-format* :alist))
	   (shasht:read-json *sbc*))

(("$schema" . "https://vega.github.io/schema/vega-lite/v5.json")
 ("description" . "A simple bar chart with embedded data.")
 ("data"
  ("values"
   . #((("a" . "A") ("b" . 28)) (("a" . "B") ("b" . 55))
       (("a" . "C") ("b" . 43)) (("a" . "D") ("b" . 91))
       (("a" . "E") ("b" . 81)) (("a" . "F") ("b" . 53))
       (("a" . "G") ("b" . 19)) (("a" . "H") ("b" . 87))
       (("a" . "I") ("b" . 52)))))
 ("mark" . "bar")
 ("encoding"
  ("x" ("field" . "a") ("type" . "nominal") ("axis" ("labelAngle" . 0)))
  ("y" ("field" . "b") ("type" . "quantitative"))))
```
That's good. A bit wordy, but certainly workable as a plotting DSL. Now try to encode it back to JSON with:
```
(shasht:write-json *)
```
and we're in the debugger with:
```
The value
  "https://vega.github.io/schema/vega-lite/v5.json"
is not of type
  LIST
   [Condition of type TYPE-ERROR]
```

**Comment by yitzchak** [comment]
The default isn't to write alists as objects, since by default I am currently using hash tables for JavaScript objects. So instead you probably need to do

```
(let ((shasht:*write-alist-as-object* t))
  (shasht:write-json *))
```

**Comment by Symbolics** [comment]
That works, thank you.
